api_views/users.py

```diff
@@ -206,3 +206,64 @@ def delete_user(username):
                 return Response(error_message_helper("User not found!"), 404, mimetype="application/json")
         else:
             return Response(error_message_helper("Only Admins may delete users!"), 401, mimetype="application/json")
+
+
+def get_user_by_email(email):
+    """
+    Retrieve user details by email address.
+    """
+    user = User.query.filter_by(email=email).first()
+    if user:
+        return jsonify(user.json()), 200
+    else:
+        return Response(error_message_helper(), 404, mimetype="application/json")
+
+
+def promote_user_to_admin(username):
+    """
+    Promote a user to admin status. Only accessible by current admins.
+    """
+    resp = token_validator(request.headers.get('Authorization'))
+    if "expired" in resp or "Invalid token" in resp:
```

Comment:
The condition 'Invalid token in resp' should be checked using string comparison with equality operator (==) rather than using 'in' operator for more precise matching.